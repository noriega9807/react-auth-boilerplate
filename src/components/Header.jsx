import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

// Font awesome
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import fontawesome from '@fortawesome/fontawesome';
import { faCoffee, faAlarmClock } from '@fortawesome/fontawesome-pro-light';


export const Header = ({startLogout}) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>
                        <span className="icon-margin-medium--right"><FontAwesomeIcon icon={faAlarmClock} /></span>
                        Flotillas
                    </h1>
                </Link>
                <button className="button button--link" onClick={startLogout}>Cerrar sesi√≥n</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);