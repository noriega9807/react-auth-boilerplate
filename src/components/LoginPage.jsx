import React from 'react';
import { connect } from 'react-redux';
import { startLogin, startRegister } from '../actions/auth';
import isEmail from 'validator/lib/isEmail';

export class LoginPage extends React.Component {

    constructor(props){
        super(props);

        this.state = {
            email: '',
            password: '',
            error: ''
        };
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    submitRegister = () => {
        if(!this.state.email || !this.state.password || !isEmail(this.state.email)){
 
            this.setState(() => ({ error: 'Por favor ingresa tus credenciales' }));
        }else{
            // clear error
            this.setState(() => ({ error: '' }));
            this.props.startRegister({
                email: this.state.email,
                password: this.state.password
            });
        }
    };

    submitLogin = () => {
        if(!this.state.email || !this.state.password || !isEmail(this.state.email)){
            //set error
            this.setState(() => ({ error: 'Por favor ingresa tus credenciales' }));
        }else{
            // clear error
            this.setState(() => ({ error: '' }));
            this.props.startLogin({
                email: this.state.email,
                password: this.state.password
            });
        }
    };

    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Login</h1>
                    <p>Ingresa tus credenciales.</p>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    {this.props.error && <p className="form__error">{this.props.error}</p>}
                    <input 
                        className="text-input text-input-login"
                        type="text"
                        placeholder="Correo"
                        autoFocus
                        value={this.state.email}
                        onChange={this.onEmailChange}
                    />
                    <input 
                        className="text-input text-input-login"
                        type="password"
                        placeholder="Contraseña"
                        value={this.state.password}
                        onChange={this.onPasswordChange}
                    />
                    <button className="button button--link button--link__login" onClick={this.submitRegister}>Registrate</button>
                    <button className="button" onClick={this.submitLogin}>Iniciar sesión</button>
                </div>
            </div>
        );
    }
    
};

const mapStateToProps = (state, props) => {
    return {
        error: state.auth.error
    };
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: (user) => dispatch(startLogin(user)),
    startRegister: (user) => dispatch(startRegister(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);