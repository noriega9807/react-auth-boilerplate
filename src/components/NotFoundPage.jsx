import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        404! <Link to="/">Go hosme</Link>
    </div>
);

export default NotFoundPage;