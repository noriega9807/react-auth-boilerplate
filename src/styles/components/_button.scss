.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--link__login {
    cursor: pointer;
    color: $blue;
    margin-bottom: 1rem;
}

.button--secondary {
    background: #888888;
}