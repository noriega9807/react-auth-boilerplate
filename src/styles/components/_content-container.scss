.content-container {
    margin: 0 auto;
    max-width: 80rem;
    padding: 0 $m-size;
}

.content-container__distribute {
    align-items: center;
    display: flex;
    justify-content: space-between;
}

.content-container__inputs {
    align-content: center;
    display: flex;
    justify-content: space-around;
    margin: $l-size;
}