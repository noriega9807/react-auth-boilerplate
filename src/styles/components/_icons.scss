// Small
.icon-margin-small--right {
    margin-right: $s-size;
}

.icon-margin-small--left {
    margin-left: $s-size;
}

.icon-margin-small--top {
    margin-top: $s-size;
}

.icon-margin-small--bottom {
    margin-bottom: $s-size;
}

// Medium
.icon-margin-medium--right {
    margin-right: $m-size;
}

.icon-margin-medium--left {
    margin-left: $m-size;
}

.icon-margin-medium--top {
    margin-top: $m-size;
}

.icon-margin-medium--bottom {
    margin-bottom: $m-size;
}

// Large
.icon-margin-large--right {
    margin-right: $l-size;
}

.icon-margin-large--left {
    margin-left: $l-size;
}

.icon-margin-large--top {
    margin-top: $l-size;
}

.icon-margin-large--bottom {
    margin-bottom: $l-size;
}