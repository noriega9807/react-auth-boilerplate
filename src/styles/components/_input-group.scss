.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-top: $m-size;
        margin-bottom: $m-size;
        justify-content: space-around;
    }
}

.input-group__item {
    margin-bottom: $s-size;
    width: 100%;

    @media (min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0;
    }
}
