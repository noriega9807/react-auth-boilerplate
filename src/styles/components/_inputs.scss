.text-input {
    border: 1px solid #cacccd;
    font-size: $font-size-large;
    font-weight: 300;
    height: 50px;
    padding: $s-size;
    width: 100%;
}

.text-input-login {
    margin-bottom: $m-size;
}

.select {
    @extend .text-input;
}


.text-area {
    @extend .text-input;
    height: 10rem;
}

.input-span--text {
    font-size: $m-size;
    align-items: center;
    display: flex;
    font-weight: bold;
    padding: $s-size;
}