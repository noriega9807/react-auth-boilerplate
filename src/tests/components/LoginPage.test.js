import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from  '../../components/LoginPage';

describe('Test LoginPage', () => {

    test('should render LoginPage', () => {
        const wrapper = shallow(<LoginPage />);

        expect(wrapper).toMatchSnapshot();

    });

    test('should call startRegister on button click', () => {
        const startRegister = jest.fn();
        const wrapper = shallow(<LoginPage startRegister={startRegister}/>);

        wrapper.setState({
            email: 'test@test.com',
            password: 'testPassword'
        });

        wrapper.find('button').at(0).simulate('click');

        expect(startRegister).toHaveBeenCalled();
    });

    test('should NOT call startRegister when empty fields', () => {
        const startRegister = jest.fn();
        const wrapper = shallow(<LoginPage startRegister={startRegister}/>);

        wrapper.setState({
            email: '',
            password: ''
        });

        wrapper.find('button').at(0).simulate('click');

        expect(wrapper.state('error')).toBe('Por favor ingresa tus credenciales');
    });

    test('should call startLogin on button click', () => {
        const startLogin = jest.fn();
        const wrapper = shallow(<LoginPage startLogin={startLogin}/>);

        wrapper.setState({
            email: 'test@test.com',
            password: 'testPassword'
        });

        wrapper.find('button').at(1).simulate('click');

        expect(startLogin).toHaveBeenCalled();
    });
});