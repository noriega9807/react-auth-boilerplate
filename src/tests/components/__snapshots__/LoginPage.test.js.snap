// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginPage should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Login
    </h1>
    <p>
      Ingresa tus credenciales.
    </p>
    <input
      autoFocus={true}
      className="text-input text-input-login"
      onChange={[Function]}
      placeholder="Correo"
      type="text"
      value=""
    />
    <input
      className="text-input text-input-login"
      onChange={[Function]}
      placeholder="Contraseña"
      type="password"
      value=""
    />
    <button
      className="button button--link button--link__login"
      onClick={[Function]}
    >
      Registrate
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Iniciar sesión
    </button>
  </div>
</div>
`;
